app/controllers/actions/IdentifierAction.scala

```diff
@@ -163,6 +162,34 @@ class AuthenticatedIdentifierAction @Inject()(taxYear: Int)
         Future.successful(Redirect(config.viewAndChangeEnterUtrUrl))
     }
   }
+
+  private def agentRecovery[A](request: Request[A],
+                               block: IdentifierRequest[A] => Future[Result],
+                               mtdItId: String)
+                              (implicit hc: HeaderCarrier): PartialFunction[Throwable, Future[Result]] = {
+    case _: NoActiveSession =>
+      logger.info(s"[AuthorisedAction][agentAuthentication] - No active session. Redirecting to ${config.viewAndChangeEnterUtrUrl}")
+      Future(Redirect(config.viewAndChangeEnterUtrUrl))
+    case _: AuthorisationException =>
+      if (config.emaSupportingAgentsEnabled) {
+        authorised(secondaryAgentPredicate(mtdItId)) {
+          block(IdentifierRequest(request, mtdItId, isAgent = true, isSecondaryAgent = true))
+        }.recoverWith {
+          case _: InternalError =>
```

Comment:
Hi Matt, may be you can test it again. I feel we dont need this  `case_: InternalError` as now you have covered it with one below with 'case _'